### ISSUE TITLE
Small problem here.

### ISSUE THREAD

**Issue by ccs21** [WonderfulToolchain/ws-cartfriend#2]
First of all, sorry for not being able to donate much.

Your project is awesome.

I cheer for you.

Small problem here.

https://youtu.be/Xet_AzuB-bY

Please watch the video.

I used beta 5.

Trying to load another game will load the wrong game or crash the system.

Like the example in the video, I run SRW Compact 2, but Cardcaptor Sakura starts...

Or, the game does not launch and the system crashes.

No matter how long I wait, it won't start.

When I restart the game I played the first time, the save data is gone.

**Comment by ccs21** [comment]
As a result of testing in various ways, it seems to be a problem with the battery.

When using other batteries, it seems to work normally.

But strange that both were new batteries.

**Comment by asiekierka** [comment]
Might be worth pointing out: selecting slot 14 is `1101` in binary; selecting slot 3 is `0010` in binary.

**Comment by ccs21** [comment]
It refers to AA batteries that are inserted into the game console body.

After a total of 4 tests using 2 alkaline batteries and 2 Li-MH batteries

One Li-MH battery had the same problem as the video.

When I checked the capacity with a battery tester, all four showed 100% output.

I'm very curious about this, but I don't think it's a software problem.

**Comment by asiekierka** [comment]
If it's a hardware problem, your best bet is to contact Flavor, who designed and sells the hardware for these cartridges. Some Flash Masta revisions had quirks of their own. I cannot do anything about hardware problems :-(

**Comment by ccs21** [comment]
The software you make is perfect.

I'm sorry for opening an issue without further testing.

It was my own support for you.

Thanks for making great software!

**Comment by asiekierka** [comment]
No need to apologize - I'd rather receive ten bug reports too many than one too few.

**Comment by asiekierka** [comment]
I have inquired about your issue further. What you might be running into is that:

* The IPS mod you have noticeably increases power consumption,
* A complex flashcart will also typically use more power than a commercial game (and remember - CartFriend uses it more intensely than the old menu).

The combination of these two factors might, hypothetically, make the single AA that powers your WonderSwan Color struggle to keep up with the power demands at times - the exact specifics would depend on brand, formulation, quality, et cetera. There's rechargeable battery mods available which provide better power delivery.

**Comment by ccs21** [comment]
thank you! 

I think that's the problem too. 
When I use the rechargeable battery mod, the game machine looks so ugly. 
But I couldn't give up on this wonderful software.
so I installed the battery mod. 

I think, for now, there are no big problems. 
I'll let you know if there are any other problems.

**Comment by ccs21** [comment]
i bother you so much. sorry.

Attach video.

https://youtu.be/ipTygwip2kw

I used a rechargeable battery mod.

I don't know the version of the Flashmasta, but it was 3D printed.

It worked just fine when I shared feedback with you yesterday.

Today, 'Mr. Driller', the symptom appeared again.

Strangely, if I take out the battery and put it back in, it works normally again.

I don't know much about software and hardware, so I can't do anything other than send you a video of the symptoms.

**Comment by asiekierka** [comment]
I have prepared a special testing build for you, which might help us figure out the issue.

[CartFriend01b6-test1.zip](https://asie.pl/files/CartFriend01b6-test1.zip)

Other than the changes scheduled for the next update of CartFriend, it adds the following features relevant to this issue:

* If communication with the flashcart fails, instead of crashing the system and causing potential corruption, CartFriend should now emit an error (a series of hexadecimal codes in the top-left corner) and stop running safely.
* A new option has been added: `Settings -> Cart AVR comm. delay`, with a default of 15 milliseconds. It might be the case that higher values fix the issue for you, but I cannot test on my Flash Masta, as it works fine for me.

In addition, could you please share the revision of Flash Masta hardware you have? (For example, mine is `4.5`).

**Comment by ccs21** [comment]
thank you for your support I'll give it a try and report back if I get any error codes.

**Comment by ccs21** [comment]
I use beta 6 and no problems so far!

Thank you for your support.

**Comment by asiekierka** [comment]
I found the issue.

"Low battery" signalization is provided via a non-maskable interrupt, while the handler was in ROM - therefore, if the "low battery" signal appeared during a flash write/erase, when ROM is inaccessible, CartFriend would crash.

Even though it was easy enough to fix (and I pushed a fix), for now I have simply removed the "low battery" signal detection, as there's already an LCD segment which states the same thing.

Either way, this has now been fixed.
